definitions/checks/system_registration.rb

```diff
@@ -10,15 +10,23 @@ class Checks::SystemRegistration < ForemanMaintain::Check
   end
 
   def run
-    if system_is_self_registerd?
+    if warn?
       raise ForemanMaintain::Error::Warn, 'System is self registered'
     else
       puts 'System is not self registered'
     end
   end
 
-  def system_is_self_registerd?
-    rhsm_hostname.casecmp(hostname).zero?
+  def system_self_registerd?
+    @result ||= rhsm_hostname.casecmp(hostname).zero?
+  end
+
+  def warn?
+    if feature(:foreman_server).present?
+      !system_self_registerd?
```

Comment:
Per Lukas's comment, a warning should be raised if the system is a **smart-proxy and self-registered**.
https://github.com/theforeman/foreman_maintain/issues/23#issuecomment-300745839